### PR TITLE
Clique rpc

### PR DESCRIPTION
**Pull request by tkstanczak**

### CODE REVIEW

File: src/Nethermind/Nethermind.Clique/CliqueBlockProducer.cs [hunk 1]
```diff
@@ -82,6 +85,23 @@ public class CliqueBlockProducer : IBlockProducer, IDisposable
...
+
+        public void CastVote(Address signer, bool vote)
+        {
+            _proposals.Add(signer, vote);
+            if(_logger.IsWarn) _logger.Warn($"Added Clique vote for {signer} - {vote}");
+        }
+        
+        public void UncastVote(Address signer)
+        {
+            if (!_proposals.ContainsKey(signer))
```
Comment (spetz) [inline src/Nethermind/Nethermind.Clique/CliqueBlockProducer.cs]: No ConcurrentDicitionary for this class?

File: src/Nethermind/Nethermind.Clique/CliqueBridge.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    public class CliqueBridge : ICliqueBridge
+    {
+        private const string CannotVoteOnNonValidatorMessage = "Cannot vote on non-validator node";
+        
+        private readonly CliqueBlockProducer _cliqueBlockProducer;
+        private readonly IBlockTree _blockTree;
+
+        public CliqueBridge(CliqueBlockProducer cliqueBlockProducer, IBlockTree blockTree)
+        {
+            if (_cliqueBlockProducer != null)
```
Comment (spetz) [inline src/Nethermind/Nethermind.Clique/CliqueBridge.cs]: Isn't it always false?

File: src/Nethermind/Nethermind.JsonRpc/Module/NullModuleProvider.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        {
+        }
+
+        public void Register<T>(IModule module) where T : IModule
+        {
+        }
+
+        public IReadOnlyCollection<ModuleInfo> GetEnabledModules()
+        {
+            return ArraySegment<ModuleInfo>.Empty;
```
Comment (spetz) [inline src/Nethermind/Nethermind.JsonRpc/Module/NullModuleProvider.cs]: Eventually, someone is using this type :D.

File: src/Nethermind/Nethermind.Clique/CliqueBridge.cs (hunk 2 above)
Comment (tkstanczak) [reply src/Nethermind/Nethermind.Clique/CliqueBridge.cs]: in this particular situation we can have clique running but not mining and then block producer is null

**Pull request closed without merging**